synedrion/src/uint/secret_unsigned.rs

```diff
@@ -0,0 +1,114 @@
+use core::ops::BitAnd;
+
+use crypto_bigint::{subtle::Choice, Bounded, Integer, Monty, NonZero};
+use zeroize::Zeroize;
+
+use super::{HasWide, SecretSigned};
+use crate::tools::Secret;
+
+/// A bounded unsigned integer with sensitive data.
+#[derive(Debug, Clone)]
+pub(crate) struct SecretUnsigned<T: Zeroize> {
+    /// bound on the bit size of the value (that is, `value < 2^bound`)
+    bound: u32,
+    value: Secret<T>,
+}
+
+impl<T> SecretUnsigned<T>
+where
+    T: Zeroize + Integer + Bounded,
+{
+    pub fn is_zero(&self) -> Choice {
+        self.value.expose_secret().is_zero()
+    }
+
+    pub fn bound(&self) -> u32 {
+        self.bound
+    }
+
+    /// Creates a new [`Bounded`] wrapper around `T`, restricted to `bound`.
+    ///
+    /// Returns `None` if the bound is invalid, i.e.:
+    /// - The bound is bigger than a `T` can represent.
+    /// - The value of `T` is too big to be bounded by the provided bound.
+    pub fn new(value: Secret<T>, bound: u32) -> Option<Self> {
+        if bound > T::BITS || value.expose_secret().bits() > bound {
+            return None;
+        }
+        Some(Self { value, bound })
+    }
+
+    pub fn add_mod(&self, rhs: &Self, modulus: &Secret<NonZero<T>>) -> Self {
+        // Note: assuming that the bit size of the modulus is not secret
```

Comment:
Move to doc-comment perhaps?

Reply:
Decided to just use `bits()` instead of `bits_vartime()` and avoid extra documentation